posts/setting-up-a-shared-library-and-seed-job-in-jenkins-1.md

```diff
@@ -0,0 +1,134 @@
+       * Leave everything else as the default configuration.
+   ![seed-config-script](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-seed-script-config-2.png)
+
+### Configure `seedJob` to use the microservice name as the job name
+We will give our job the name of the microservice we plan to build (`poc-micro`). In order to do this we will need to add a `String parameter` to the `seedJob` that will be used inside of `seed.groovy`.
+   1. Navigate to `Dashboard` > select `seedJob` > select `Configure` .
+   2. Select `This project is parameterized` > select `Add Parameter` > select `String Parameter`.
+   3. Enter `jobName` in `Name` field.
+   4. In the `Description` field, enter `The name of your repo (e.g. poc-micro)`.
+   ![jenkins seed job configuration](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-seed-add-jobName.png)
+
+## Goal 2
+We are using the [`microservice-pipelines`](https://github.com/kcrane3576/microservice-pipelines) github repository. This repository will be used to store our `seed` code. In Part 2 of this series, we will include our Shared Library code in this repository.
+
+  1. Inside of the `microservice-pipelines` repository, we have created a directory `dsl` with `seed.groovy` inside.
+      * [`microservice-pipelines/dsl/seed.groovy`](https://github.com/kcrane3576/microservice-pipelines/blob/master/dsl/seed.groovy).
+  2. Below are the contents of `seed.groovy`. 
+      * We create a simple Freestyle Job and use the `String Param` named `jobName` from `seedJob` to name our Freestyle job. 
+```groovy
+job(jobName) {
+    description("A simple Freestyle Job created from seed.groovy")
+}
+  ```
+   
+##  Goal 3
+Now that we have our `seedJob` setup to read in `seed.groovy` from our github `microservice-pipelines` repository, we are ready to trigger our `seedJob` to create a Freestyle job with `jobName`.
+
+### Running the `seedJob`
+  1. Navigate to `Dashboard` > select `seedJob` > select `Build Now`.
+  2. Under `Build History`, select the top red circle.
+  3. This will take you to the `Console Output`.
+     * The job **failed**.
+     * Due to [Script Security](https://github.com/jenkinsci/job-dsl-plugin/wiki/Script-Security), this will happen every time you change `seed.groovy`. The [Script Security Plugin](https://plugins.jenkins.io/script-security) is integrated with the `Job DSL` plugin and the Script Security Plugin is set up with a set of scripts that are pre approved for use. Since this is a new script, it will require an admin approval to use. 
+     ![run failure](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-seed-run-failure.png)
+      
+### Approving our `seed.groovy` script
+  1. We need to tell Jenkins it is ok to use this script.
+    * Navivage to `Dashboard` > `Manage Jenkins` > `In-process Script Approval`.
```

Comment:
navigate

Reply:
The latest version has this and other type-o's corrected